plugins/package-managers/maven/src/main/kotlin/utils/DependencyTreeParser.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2025 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.plugins.packagemanagers.maven.utils
+
+import java.io.InputStream
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.decodeToSequence
+
+import org.apache.maven.project.MavenProject
+
+import org.eclipse.aether.artifact.DefaultArtifact
+import org.eclipse.aether.graph.DefaultDependencyNode
+import org.eclipse.aether.graph.Dependency
+import org.eclipse.aether.graph.DependencyNode
+import org.eclipse.aether.repository.RemoteRepository
+
+/** The [Json] instance to use for parsing of dependency trees in JSON format. */
+internal val JSON = Json { ignoreUnknownKeys = true }
+
+/**
+ * A data class to represent a node in the JSON output generated by the Maven Dependency Plugin.
+ */
+@Serializable
+internal data class DependencyTreeMojoNode(
+    val groupId: String,
+    val artifactId: String,
+    val version: String,
+    val type: String,
+    val scope: String,
+    val classifier: String,
+    val children: List<DependencyTreeMojoNode> = emptyList()
+) {
+    /** An ID that matches the internal IDs generated for Maven projects. */
+    val projectId: String
+        get() = "$groupId:$artifactId:$type:$version"
```

Comment:
Can't this be just `val projectId = "$groupId:$artifactId:$type:$version"`, as there is no need to re-evaluate?